packages/vips/src/index.ts

```diff
@@ -199,31 +189,14 @@ export async function resizeImage(
 
 	image.onProgress = onProgress;
 
-	const { width } = image;
-
-	// Using getTypeof acts an isset check.
-	const numberOfFrames =
-		supportsAnimation( type ) &&
-		image.getTypeof( 'n-pages' ) &&
-		! resize.crop
-			? image.getInt( 'n-pages' )
-			: 1;
-	const height = image.height / numberOfFrames;
-	const isAnimated = numberOfFrames > 1;
-
-	// To preserve all frames when cropping.
-	if ( isAnimated ) {
-		thumbnailOptions.option_string = '[n=-1]';
-	}
```

Comment:
I think this still needs to be kept for the cropping, but it can probably be simplified like this:

```ts
const isAnimated = pageHeight !== image.height;

// To preserve all frames when cropping.
if ( isAnimated ) {
	thumbnailOptions.option_string = '[n=-1]';
}
```

Reply:
Hmm since tests are passing, I‘ll need to look into this more and maybe add a new test 🧐

Reply:
This logic was consolidated with the if-statement above.
https://github.com/swissspidy/media-experiments/blob/4c5a09c81a3fc4031e9abf3a2ff84f2d06bd819d/packages/vips/src/index.ts#L177-L183

Previously `isAnimated` was always `false` when cropping due to:
https://github.com/swissspidy/media-experiments/blob/071c634e8f14d13ef22a2407112b2854ee9707ff/packages/vips/src/index.ts#L208

Cropping images with multiple pages can be somewhat non-trivial, sharp uses the following function for this:
https://github.com/lovell/sharp/blob/v0.33.4/src/operations.cc#L379